File: samples.vb/AdventureWorksLT.SqlServer/AdventureWorksLT.DbInit/DataSetExtensions.vb
#If Not DbInit Then

Imports System.Runtime.CompilerServices
Imports DevZest.Data

Module DataSetExtensions
    <Extension()>
    Public Function AddRows(Of T As {Class, IEntity, New})(ByVal dataSet As DataSet(Of T), ByVal count As Integer) As DataSet(Of T)
        For i = 1 To count
            dataSet.AddRow()
        Next
        Return dataSet
    End Function
End Module

#End If

File: samples/AdventureWorksLT.SqlServer/AdventureWorksLT.DbInit/DataSetExtensions.cs
#if !DbInit
using DevZest.Data.Primitives;

namespace DevZest.Data
{
    public static class DataSetExtensions
    {
        public static DataSet<T> AddRows<T>(this DataSet<T> dataSet, int count)
            where T : class, IEntity, new()
        {
            for (int i = 0; i < count; i++)
                dataSet.AddRow();

            return dataSet;
        }
    }
}
#endif